.github/workflows/codecov.yml

```diff
@@ -0,0 +1,349 @@
+name: Prune Codecov .coveragerc
+on: # yamllint disable-line rule:truthy
+  pull_request:
+    branches:
+      - dev
+    paths:
+      - .github/workflows/codecov.yml
+      - script/check_coveragerc.py
+  schedule:
+    # Every Monday at 4am UTC
+    - cron: 0 4 * * 1
+  workflow_dispatch:
+
+env:
+  GH_TOKEN: ${{ github.token }}
+  BRANCH_NAME: "prune_coveragerc_1"
+  USER_EMAIL: "41898282+github-actions[bot]@users.noreply.github.com"
+  USER_NAME: "github-actions[bot]"
+  PR_BODY: |
+    *This is an automatically generated pull request created by /.github/workflows/codecov.yml*
+
+    Hi! When analysing '.coveragerc', I found that some files were omitted from pytest coverage checks that didn't need to be.
+
+    This usually happens when those files already have a high or 100% coverage.
+
+    I created a branch to remove these entries. Please review and merge the branch if you agree.
+
+    Next time this workflow runs, it will automatically update the branch under the same PR, or recreate the branch or PR if they have been deleted.
+
+  CACHE_VERSION: 5
+  UV_CACHE_VERSION: 1
+  MYPY_CACHE_VERSION: 8
+  HA_SHORT_VERSION: "2024.5"
+  DEFAULT_PYTHON: "3.12"
+  ALL_PYTHON_VERSIONS: "['3.12']"
+  # 10.3 is the oldest supported version
+  # - 10.3.32 is the version currently shipped with Synology (as of 17 Feb 2022)
+  # 10.6 is the current long-term-support
+  # - 10.6.10 is the version currently shipped with the Add-on (as of 31 Jan 2023)
+  # 10.10 is the latest short-term-support
+  # - 10.10.3 is the latest (as of 6 Feb 2023)
+  # 10.11 is the latest long-term-support
+  # - 10.11.2 is the version currently shipped with Synology (as of 11 Oct 2023)
+  # mysql 8.0.32 does not always behave the same as MariaDB
+  # and some queries that work on MariaDB do not work on MySQL
+  MARIADB_VERSIONS: "['mariadb:10.3.32','mariadb:10.6.10','mariadb:10.10.3','mariadb:10.11.2','mysql:8.0.32']"
```

Comment:
Please check if all env variables are required, and if not remove the unused ones.
I expect the at least the db variables can be removed

Reply:
Done